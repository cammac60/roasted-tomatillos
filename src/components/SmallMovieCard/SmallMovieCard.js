import React, { Component } from 'react';
import './SmallMovieCard.scss'
import { Redirect } from 'react-router-dom';

class SmallMovieCard extends Component {
  constructor(props) {
    super(props)
    this.state = {
      isRedirected: false
    }
  }

  redirect = () => {
    this.setState({isRedirected: true})
  }

  render() {
    const { id, img, title, rate } = this.props

    const headerStyles = {
      backgroundImage: `url("${img}")`,
      backgroundSize: '100%',
      backgroundPosition: 'center',
      width: '230px',
      height: '300px',
    }

    return (
      (this.state.isRedirected)
        ? <Redirect to={`/movies/${id}`} />
        : <section
          className="small-movie-card"
          onClick={this.redirect}>
          <header style={headerStyles}></header>
          <p className="rate-number">{ rate }</p>
          <div className="title-block">
            <p>{ title }</p>
          </div>
        </section>
    );
  }
}

export default SmallMovieCard;
