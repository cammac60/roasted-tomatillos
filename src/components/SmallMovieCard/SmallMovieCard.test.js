import React from 'react';
import { shallow } from 'enzyme';
import SmallMovieCard from './SmallMovieCard';

describe("SmallMovieCard", () => {
  let card, mockProps;

  beforeEach(() => {
    mockProps = {
      id: 1,
      img: 'https://some.url/img.jpg',
      title: 'Jumanji: The Next Level',
      rate: 4
    }

    card = shallow(
      <SmallMovieCard { ...mockProps } />
    )
  });

  it("should match snapshot with all data passed in correctly", () => {
    expect(card).toMatchSnapshot();
  });

  it("should match snapshot with redirect router if redirect state is true", () => {
    card.setState({isRedirected: true});
    expect(card).toMatchSnapshot();
  });

  it("should change redirect state to true after click on card", () => {
    expect(card.state('isRedirected')).toEqual(false);

    card.find('section').simulate('click');

    expect(card.state('isRedirected')).toEqual(true);
  });
});
