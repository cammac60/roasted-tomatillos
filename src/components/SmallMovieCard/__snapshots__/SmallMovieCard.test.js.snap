// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallMovieCard should match snapshot with all data passed in correctly 1`] = `
<section
  className="small-movie-card"
  onClick={[Function]}
>
  <header
    style={
      Object {
        "backgroundImage": "url(\\"https://some.url/img.jpg\\")",
        "backgroundPosition": "center",
        "backgroundSize": "100%",
        "height": "300px",
        "width": "230px",
      }
    }
  />
  <p
    className="rate-number"
  >
    4
  </p>
  <div
    className="title-block"
  >
    <p>
      Jumanji: The Next Level
    </p>
  </div>
</section>
`;

exports[`SmallMovieCard should match snapshot with redirect router if redirect state is true 1`] = `
<Redirect
  to="/movies/1"
/>
`;
